frontend/catalyst/passes/builtin_passes.py

```diff
@@ -848,9 +849,13 @@ def circuit():
 
     if QJIT.mlir is not None:
         # aot mode
-        raw_result = _quantum_opt(
-            ("--pass-pipeline", "builtin.module(ppm-specs)"), [], stdin=QJIT.mlir_opt
-        )
+        new_options = copy.copy(QJIT.compile_options)
+        if new_options.pipelines is None:
+            raise CompileError("No pipeline found")
+        new_options.pipelines[0][1].append("ppm-specs")  # add ppm-spec pass to existing pipeline
```

Comment:
```suggestion
        new_options.pipelines[-1][-1].append("ppm-specs")  # add ppm-spec pass to existing pipeline
```
I think we want to append the `ppm-specs` at the end of the pipeline. Would you check if that works?

Reply:
Well I specifically chose the first pipeline because that is the one that runs user passes (i.e. the ppm passes).

For the second access, `-1` is the same as `1` because the structure of a pipeline is a `tuple(str, list)`. Using `-1` doesn't semantically add anything imo.

Reply:
Probably good to add a comment about those magic numbers. Structured binding is also a great way to give tuple elements meaning, so something like
```py
_, pass_list = new_options.pipelines[0]  # first pipeline runs the user passes
pass_list.append(...)
```

Reply:
Thanks David. Added it here https://github.com/PennyLaneAI/catalyst/commit/db3c39f74c4b3b4b2cf21d222a0b7291283bc4af